web/modules/custom/xlsx/src/Entity/XlsxEntity.php

```diff
@@ -116,21 +116,33 @@ public function isCron() {
    */
   public function shouldExecuteOnCron() {
     $mapping = $this->getMapping();
-    if ($this->isCron()) {
-      $inteval = (int) $mapping['cron_frequency'];
-      if (empty($interval)) {
+    $inteval = (int) $mapping['cron_frequency'];
+    if ($this->isCron() && !empty($inteval)) {
+      $last_cron_run = \Drupal::state()->get('xlsx.last_cron_run');
+      $current_time = \Drupal::time()->getCurrentTime();
+      if (empty($last_cron_run[$this->id()])) {
         return TRUE;
       }
-      else {
-        $last_cron_run = \Drupal::state()->get('xlsx.last_cron_run');
-        $current_time = \Drupal::time()->getCurrentTime();
-        if (empty($last_cron_run[$this->id()]) || ($current_time >= $last_cron_run[$this->id()])) {
-          $last_cron_run[$this->id()] = $current_time + $interval;
-          \Drupal::state()->set('xlsx.last_cron_run', $last_cron_run);
-          return TRUE;
-        }
+      else if (!empty($last_cron_run[$this->id()]) && ($current_time < $last_cron_run[$this->id()])) {
```

Comment:
You don't need the `else if`.  If `$last_cron_run[$this->id()]` is empty here you would have already returned TRUE on 124, so we know it's not empty by 126.

So you just need `if ($current_time < $last_cron_run[$this->id()])`.

Reply:
@sleepingmonk what do you think about this instead; `if (empty($last_cron_run[$this->id()]) || ($current_time < $last_cron_run[$this->id()])) {`

Reply:
@sleepingmonk This is fixed https://github.com/JudicialCouncilOfCalifornia/trialcourt/pull/2741/commits/7183247bdc48186e4ba706cd69c798df071d06c1